core/scratch_msgs.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2018 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Scratch Messages singleton, with function to override Blockly.Msg values.
+ * @author chrisg@media.mit.edu (Chris Garrity)
+ */
+'use strict';
+
+/**
+ * Name space for the Msg singleton.
+ * Msg gets populated in the message files.
+ */
+goog.provide('Blockly.ScratchMsgs');
+goog.require('Blockly.Msg');
+
+/**
+ * Change the Blockly.Msg strings to a new Locale
+ * Does not exist in Blockly, but needed in scratch-blocks
+ * @param {string} locale E.g., 'de', or 'zh-tw'
+ * @package
+ */
+Blockly.ScratchMsgs.setLocale = function(locale) {
+  if (Object.keys(Blockly.ScratchMsgs.locales).includes(locale)) {
+    Blockly.Msg = Blockly.ScratchMsgs.locales[locale];
```

Comment:
I think this will work in most places. It will not work in dropdowns.  See https://github.com/google/blockly/issues/1895

Reply:
@AnmAtAnm is there a particular type of dropdown that shouldn't work? In my vertical playground the looks effect dropdown on the `change %1 effect by %2` updates when I switch languages as does the dropdown on the `go to %1` block. However, I've noticed that we're missing some dropdown strings entirely (previous backdrop), and maybe they will be a problem.

Reply:
I'm presuming you blocks and static dropdowns use the same JSON-based loading mechanism as Blockly.  Dynamic drop downs and `block.init()` initialized fields should not have this error. There is a possibility your team has fixed the underlying bug, but I haven't heard anything like that. It comes down to how the options table is centralized for the specific dropdown of a particular type of block (so the options can be programmatically updated everywhere), and in that process the option message is dereference once, rather than per-block.

Reply:
I think all our dropdowns are initialized in the `init()` function, so maybe it's moot for scratch-blocks. There are definitely some issues with dynamic dropdowns, but they're actually getting initialized in the gui and that is what I'm working on next.